widgets/common/src/filters/RuleOperatorDropdown.tsx

```diff
@@ -1,28 +1,52 @@
 import { camelCase } from 'lodash';
-import type { FunctionComponent } from 'react';
+import { FunctionComponent, useEffect, useState } from 'react';
 import type { DropdownItemProps } from 'semantic-ui-react/dist/commonjs/modules/Dropdown/DropdownItem';
+
 import { i18nPrefix } from './consts';
+import { FilterRuleRowType, LabelsFilterRuleOperators, AttributesFilterRuleOperators } from './types';
+import type { FilterRuleOperator } from './types';
+
+interface RuleOperatorDropdownProps {
+    ruleType: FilterRuleRowType;
+    onChange: (value: FilterRuleOperator) => void;
+    value: FilterRuleOperator;
+}
+
+function getDropdownOptions(operators: string[]) {
+    const { i18n } = Stage;
+    return operators.map(
+        (operator): DropdownItemProps => ({
+            text: i18n.t(`${i18nPrefix}.operatorsLabels.${camelCase(operator as string)}`),
+            value: operator
+        })
+    );
+}
 
-const { i18n } = Stage;
-const operatorsOptions = [
-    'any_of',
-    'not_any_of',
-    'is_null',
-    'is_not_null',
-    'contain',
-    'not_contain',
-    'start_with',
-    'end_with'
-].map(
-    (operator): DropdownItemProps => ({
-        text: i18n.t(`${i18nPrefix}.operatorsLabels.${camelCase(operator)}`),
-        value: operator
-    })
-);
-
-const RuleOperatorDropdown: FunctionComponent = () => {
+const RuleOperatorDropdown: FunctionComponent<RuleOperatorDropdownProps> = ({ ruleType, onChange, value }) => {
     const { Dropdown } = Stage.Basic;
 
-    return <Dropdown search selection name="ruleOperator" options={operatorsOptions} />;
+    const [options, setOptions] = useState([] as DropdownItemProps[]);
```

Comment:
Digression:

I've been always using the following:

```suggestion
    const [options, setOptions] = useState<DropdownItemProps[]>([]);
```

That handles a bit more options, e.g. if you do not provide the default `[]` value, then `options` will be `DropdownItemProps[] | undefined` :slightly_smiling_face: 

Both are valid, I just wanted to show another way to do the same thing, one that can save you some typing in the future :slightly_smiling_face:

Reply:
Nice, I wasn't aware of it 👍🏻  Will change it.